sdk/go/api_esc_extensions.go

```diff
@@ -0,0 +1,287 @@
+package esc
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"gopkg.in/ghodss/yaml.v1"
+)
+
+type EscClient struct {
+	rawClient *RawAPIClient
+	EscAPI    *EscAPIService
+}
+
+func NewClient(cfg *Configuration) *EscClient {
+	client := &EscClient{rawClient: NewRawAPIClient(cfg)}
+	client.EscAPI = client.rawClient.EscAPI
+	return client
+}
+
+func (c *EscClient) ListEnvironments(ctx context.Context, org string, continuationToken *string) (*OrgEnvironments, error) {
+	request := c.EscAPI.ListEnvironments(ctx, org)
+	if continuationToken != nil {
+		request = request.ContinuationToken(*continuationToken)
+	}
+
+	envs, _, err := request.Execute()
+	return envs, err
+}
+
+func (c *EscClient) GetEnvironment(ctx context.Context, org, envName string) (*EnvironmentDefinition, string, error) {
+	env, resp, err := c.EscAPI.GetEnvironment(ctx, org, envName).Execute()
+	if err != nil {
+		return nil, "", err
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, "", err
+	}
+
+	return env, string(body), nil
+}
+
+func (c *EscClient) OpenEnvironment(ctx context.Context, org, envName string) (*OpenEnvironment, error) {
+	openInfo, _, err := c.EscAPI.OpenEnvironment(ctx, org, envName).Execute()
+	return openInfo, err
+}
+
+func (c *EscClient) ReadOpenEnvironment(ctx context.Context, org, envName, openEnvID string) (*Environment, map[string]any, error) {
+	env, _, err := c.EscAPI.ReadOpenEnvironment(ctx, org, envName, openEnvID).Execute()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	propertyMap := *env.Properties
+	for k, v := range propertyMap {
+		v.Value = mapValues(v.Value)
+		propertyMap[k] = v
+	}
```

Comment:
This is mutating the property map in the environment. Is that intentional?

Reply:
Yes, in order to make it more strongly typed.  The first level is of type Value (strongly typed).  This iterates and makes child values type Value too (they start as any)